solutions/usaco-592.mdx

```diff
@@ -69,4 +74,65 @@ int main() {
 ```
 
 </CPPSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class angrycowsB {
+	static int N, A = 0;
+	static int[] positions;
+	public static void main(String[] args) throws Exception {
+		Kattio io = new Kattio("angry");
+
+		N = io.nextInt();
+		positions = new int[N];
+
+		for (int i = 0; i < N; i++) {
+			positions[i] = io.nextInt();
+		}
+
+		Arrays.sort(positions);
+
+		for (int i = 0; i < N; i++) {
+			//Computing answer
+			A = Math.max(A, getNum(i,-1) + getNum(i, 1) + 1);
+		}
+
+		io.println(A);
+		io.close();
+	}
+
+	public static int getNum(int start, int direction) {
+		int radius = 1;
+		int prev = start;
+
+		while(true) {
+			int next = prev;
+			//Getting the furthest explosion index without exceeding radius
+			while(next + direction >= 0 && next + direction < N 
```

Comment:
wait why put spaces after control flow statements up there but not down here
![image](https://user-images.githubusercontent.com/55369003/127077045-4ed28b3f-ad17-47bc-ba1d-6ecbff03a82d.png)